### ISSUE TITLE
IPv6 tests are failing since Oct 18

### ISSUE THREAD

**Issue by hakman** [kubernetes/kops#16049]
/kind bug

**1. What `kops` version are you running? The command `kops version`, will display
 this information.**
CI

**2. What Kubernetes version are you running? `kubectl version` will print the
 version if a cluster is running or provide the Kubernetes version specified as
 a `kops` flag.**
Stable

**3. What cloud provider are you using?**
AWS

**4. What commands did you run?  What is the simplest way to reproduce this issue?**
https://testgrid.k8s.io/kops-network-plugins#kops-aws-cni-calico-ipv6
https://testgrid.k8s.io/kops-network-plugins#kops-aws-cni-cilium-ipv6

**5. What happened after the commands executed?**
Cluster doesn't come up because cloud-init doesn't finish on control-plane nodes, so NodeUp doesn't run.
It is odd that regular nodes run cloud-init just fine, but they are stuck at trying to connect to the masters.

**6. What did you expect to happen?**

**7. Please provide your cluster manifest. Execute
  `kops get --name my.example.com -o yaml` to display your cluster manifest.
  You may want to remove your cluster name and other sensitive information.**

```yaml
```

**8. Please run the commands with most verbose logging by adding the `-v 10` flag.
  Paste the logs into this report, or in a gist and provide the gist link here.**

**9. Anything else do we need to know?**
There are no kOps or test-infra changes that would account for the sudden failure on Oct 18, which makes it very odd.

**Comment by hakman** [comment]
CC @johngmyers @justinsb

**Comment by justinsb** [comment]
Oddly it looks like the control plane nodes failed to start our script / nodeup.  On the worker nodes, everything starts as normal.  On the control-plane nodes, it looks like it didn't even try to run the script.

**Comment by hakman** [comment]
> Oddly it looks like the control plane nodes failed to start our script / nodeup.  On the worker nodes, everything starts as normal.  On the control-plane nodes, it looks like it didn't even try to run the script.

Exactly, pretty suspicios, like an AWS related change/bug.

**Comment by justinsb** [comment]
I'm not entirely sure I understand what's going on, but it might be related to the CDN

From an IPv6 AWS instance:

```
ubuntu@i-0ed623b3c82c97e7d:~$ curl -I https://cdn.dl.k8s.io/release/v1.28.3/bin/linux/amd64/kubectl
HTTP/2 200 
expires: Fri, 20 Oct 2023 18:18:28 GMT
last-modified: Wed, 18 Oct 2023 14:34:51 GMT
etag: "f5d9b873b9da275032807a196a0acc44"
content-type: application/octet-stream
cache-control: private, no-store
accept-ranges: bytes
date: Sat, 28 Oct 2023 14:51:49 GMT
via: 1.1 varnish
age: 682401
x-served-by: cache-bfi-krnt7300045-BFI
x-cache: HIT
x-cache-hits: 1
access-control-allow-origin: *
content-length: 49872896

ubuntu@i-0ed623b3c82c97e7d:~$ curl -I https://cdn.dl.k8s.io/release/v1.28.3/bin/linux/arm64/kubectl
HTTP/2 404 
content-type: application/xml; charset=UTF-8
expires: Wed, 04 Oct 2023 09:50:01 GMT
cache-control: private, no-store
accept-ranges: bytes
date: Sat, 28 Oct 2023 14:51:56 GMT
via: 1.1 varnish
age: 2091715
x-served-by: cache-bfi-krnt7300083-BFI
x-cache: HIT
x-cache-hits: 3
access-control-allow-origin: *
content-length: 220
```

From my machine:

```
> curl -6 -I https://cdn.dl.k8s.io/release/v1.28.3/bin/linux/arm64/kubectl
HTTP/2 200 
expires: Sat, 28 Oct 2023 15:49:15 GMT
cache-control: public, max-age=3600
last-modified: Wed, 18 Oct 2023 14:34:51 GMT
etag: "8b797ec313138e9a1a6c560074b5f5a5"
content-type: application/octet-stream
accept-ranges: bytes
date: Sat, 28 Oct 2023 14:55:13 GMT
via: 1.1 varnish
age: 358
x-served-by: cache-pdk-kpdk1780105-PDK
x-cache: HIT
x-cache-hits: 1
access-control-allow-origin: *
content-length: 47710208

 > curl -4 -I https://cdn.dl.k8s.io/release/v1.28.3/bin/linux/arm64/kubectl
HTTP/2 200 
expires: Sat, 28 Oct 2023 15:49:15 GMT
cache-control: public, max-age=3600
last-modified: Wed, 18 Oct 2023 14:34:51 GMT
etag: "8b797ec313138e9a1a6c560074b5f5a5"
content-type: application/octet-stream
accept-ranges: bytes
date: Sat, 28 Oct 2023 14:55:20 GMT
via: 1.1 varnish
age: 365
x-served-by: cache-pdk-kpdk1780094-PDK
x-cache: HIT
x-cache-hits: 1
access-control-allow-origin: *
content-length: 47710208

```

Oddly though:

```
> curl  https://cdn.dl.k8s.io/release/v1.28.3/bin/linux/arm64/kubectl
<?xml version='1.0' encoding='UTF-8'?><Error><Code>NoSuchKey</Code><Message>The specified key does not exist.</Message><Details>No such object: kubernetes-release/release/v1.28.3/bin/linux/arm64/kubectl</Details></Error>
```

But I think it exists, I can `gsutil ls gs://kubernetes-release/release/v1.28.3/bin/linux/arm64/kubectl` etc.

Note the creation time, it corresponds to when our tests started failing, so I think it might be related (I assume we released 1.28.3 on this date):

```
> gsutil ls -L gs://kubernetes-release/release/v1.28.3/bin/linux/arm64/kubectl
gs://kubernetes-release/release/v1.28.3/bin/linux/arm64/kubectl:
    Creation time:          Wed, 18 Oct 2023 14:34:51 GMT
    Update time:            Wed, 18 Oct 2023 14:34:51 GMT
    Storage class:          STANDARD
    Content-Length:         47710208
    Content-Type:           application/octet-stream
    Metadata:               
        goog-reserved-file-mtime:1697631422
    Hash (crc32c):          0El0zg==
    Hash (md5):             i3l+wxMTjpoabFYAdLX1pQ==
    ETag:                   CKG8+dXo/4EDEAE=
    Generation:             1697639691083297
    Metageneration:         1
    ACL:                    []
TOTAL: 1 objects, 47710208 bytes (45.5 MiB)
```

I think we do only download kubectl on control plane nodes, so this could match.  But I don't understand why IPv6 (in some regions) would fail.  Unless maybe something went wrong with the S3 mirror to that location (though we should fall back I thought) - I'll check that.

**Comment by justinsb** [comment]
Ah, this seems like a known issue: https://github.com/kubernetes/k8s.io/issues/6020

It's possible that I just got unlucky on this particular run, and that there's a different issue affecting the tests, as I would expect to see some log messages in this case.

**Comment by hakman** [comment]
We don't even get to running the kops-configuration service, so probably unrelated.

**Comment by justinsb** [comment]
Yeah, you're right.  I'm working on the theory now that we've exhausted some resource in the test account; we are failing to delete some ipv6 vpcs in the janitor: https://storage.googleapis.com/kubernetes-jenkins/logs/maintenance-ci-aws-janitor/1718347447807250432/build-log.txt

**Comment by justinsb** [comment]
I looked in the test account and there were some egress-only-internet-gateways blocking VPC deletion.  I'm surprised that would fail "silenty" so I added some dumps in #16069.  I also cleaned up the test account (perhaps that wasn't the best plan because then we don't get the dump information) but it does look things are still failing.

**Comment by justinsb** [comment]
The tests seem to have self-healed so I think your theory @hakman about it being something odd in AWS was likely correct.  I don't think the janitor changes would have made a difference, as I cleaned these up anyway.  I don't see any difference when it went green - same image, same git sha of kOps, same version of kube. 

I guess we should watch for another day or two before declaring "success" - and we did find some interesting things along the way!
